sysconfig/cloudinit.go

```diff
@@ -76,21 +80,50 @@ func installCloudInitCfg(src, targetdir string) error {
 	return nil
 }
 
-// TODO:UC20: - allow cloud.conf coming from the gadget
-//            - think about if/what cloud-init means on "secured" models
+// installUnifiedCloudInitCfg installs a single cloud-init config file to the
+// cloud config dir as "ubuntu-core-cloud_91.cfg".
+func installUnifiedCloudInitCfg(src, targetdir string) error {
+	ubuntuDataCloudCfgDir := filepath.Join(ubuntuDataCloudDir(targetdir), "cloud.cfg.d/")
+	if err := os.MkdirAll(ubuntuDataCloudCfgDir, 0755); err != nil {
+		return fmt.Errorf("cannot make cloud config dir: %v", err)
+	}
+
+	configFile := filepath.Join(ubuntuDataCloudCfgDir, "ubuntu-core-cloud_91.cfg")
```

Comment:
the priority needs to go in front, also curtin uses 50- so maybe we should use 49- ?  something maybe to discuss with the server team.

I don't think we need "cloud" in the name,  we are in cloud/cloud.cfg.d already

Reply:
ah yes sorry I forgot that the priority needs to be in front, I was thinking though that the gadget should take precedence over curtin ?

Reply:
thinking about this further, we have also the option (especially when we add filtering) to renumber the files we copy from seed, at that point we have more freedom to chose a number that makes sense only related to what is in core* and our purposes